ps3-stable/cell-add-spu-shutdown-method.diff

old:
patches/other/spufs-fix-ctx-lifetimes.diff
patches/other/spufs-always-release-mapping-lock.diff

---
 arch/powerpc/platforms/cell/spu_base.c |   15 ++++++++++++++-
 1 file changed, 14 insertions(+), 1 deletion(-)

--- a/arch/powerpc/platforms/cell/spu_base.c
+++ b/arch/powerpc/platforms/cell/spu_base.c
@@ -565,8 +565,21 @@ int spu_irq_class_1_bottom(struct spu *s
 	return ret;
 }
 
+static int spu_shutdown(struct sys_device *sysdev)
+{
+	struct spu *spu = container_of(sysdev, struct spu, sysdev);
+
+	// what else here???
+
+	spu_free_irqs(spu);
+	spu_destroy_spu(spu);
+	kfree(spu);
+	return 0;
+}
+
 struct sysdev_class spu_sysdev_class = {
-	set_kset_name("spu")
+	set_kset_name("spu"),
+	.shutdown = spu_shutdown,
 };
 
 int spu_add_sysdev_attr(struct sysdev_attribute *attr)
